### DISCUSSION TITLE
When updating Sonoff Basic from Tasmota 7.2 to 8.5.1 (or even to 13.1.0) physical button stops working

### DISCUSSION THREAD

**Discussion by kmgalanakis** [arendst/Tasmota#19483] [Q&A] [answered]
### PROBLEM DESCRIPTION
Yesterday I updated my Sonoff Basic from 7.2 to 8.5.1 (and then to 9.1 and eventually 13.1 to get auto-discovery on HA) and I realized that even though everything was working properly, the physical button was no longer working. 

In order to get the button back to a working state I have to downgrade again to 7.2. Any version after 7.2 disables the button.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff Basic
- [x] Tasmota binary firmware version number used: 8.5.1, 9.1, 13.1
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
13:18:12.992 MQT: stat/sonoff3/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
13:18:13.201 MQT: stat/sonoff3/RESULT = {"Module":{"1":"Sonoff Basic"}}
13:18:13.457 MQT: stat/sonoff3/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
13:19:21.305 MQT: stat/sonoff3/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
13:19:21.518 MQT: stat/sonoff3/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
13:19:21.769 MQT: stat/sonoff3/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
13:19:45.407 MQT: stat/sonoff3/STATUS = {"Status":{"Module":1,"DeviceName":"Sonoff 3","FriendlyName":["Sonoff 3"],"Topic":"sonoff3","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":1,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
13:19:45.444 MQT: stat/sonoff3/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://thehackbox.org/tasmota/release/tasmota.bin","RestartReason":"Software/System restart","Uptime":"0T00:04:08","StartupUTC":"2023-09-09T12:15:37","Sleep":50,"CfgHolder":4617,"BootCount":7,"BCResetTime":"2023-09-08T19:35:43","SaveCount":15,"SaveAddress":"F6000"}}
13:19:45.477 MQT: stat/sonoff3/STATUS2 = {"StatusFWR":{"Version":"13.1.0(lite)","BuildDateTime":"2023-08-15T07:59:31","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"345/699"}}
13:19:45.496 MQT: stat/sonoff3/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["KMGHome",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008029","0105C80001000600003C5A00002800000000","00000080","00006000","00004000","00000000"]}}
13:19:45.525 MQT: stat/sonoff3/STATUS4 = {"StatusMEM":{"ProgramSize":494,"Free":508,"Heap":27,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"14405E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","0F000782","04108001","00000002","00000000","00000000","00000000","00000000","04000000","00000080"],"Drivers":"1,2,4,9,10,12,16,20,21,62","Sensors":""}}
13:19:45.557 MQT: stat/sonoff3/STATUS5 = {"StatusNET":{"Hostname":"sonoff3-2105","IPAddress":"192.168.1.132","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"DC:4F:22:09:C8:39","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
13:19:45.580 MQT: stat/sonoff3/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.104","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_09C839","MqttUser":"kmgalanakis","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
13:19:45.604 MQT: stat/sonoff3/STATUS7 = {"StatusTIM":{"UTC":"2023-09-09T12:19:45","Local":"2023-09-09T13:19:45","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"06:18","Sunset":"19:16"}}
13:19:45.623 MQT: stat/sonoff3/STATUS10 = {"StatusSNS":{"Time":"2023-09-09T13:19:45"}}
13:19:45.633 MQT: stat/sonoff3/STATUS11 = {"StatusSTS":{"Time":"2023-09-09T13:19:45","Uptime":"0T00:04:08","UptimeSec":248,"Vcc":3.489,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"KMGHome","BSSId":"D4:86:60:BB:77:69","Channel":13,"Mode":"11n","RSSI":100,"Signal":-38,"LinkCount":1,"Downtime":"0T00:00:17"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
No support or bug fixing for such very old versions. Update to llatest release Version.
If the issue is still there, you can reopen.

**Comment by kmgalanakis** [comment]
@Jason2866 as I mentioned in the description, the issue is still there even for 13.1.0. I only mentioned 8.5.1 because it was the first version that broke the physical button functionality.

I'm reopening the issue.

**Comment by kmgalanakis** [comment]
@Jason2866 apparently I cannot reopen the issue, could you please do it for me?
 
**Edit:** I also modified the issue title to reflect that the issue is still present on 13.1.0.

**Comment by Jason2866** [comment]
Moving to discussions, since it is not an issue with current version

**Comment by kmgalanakis** [comment]
@Jason2866 I'm struggling to understand how the fact that an update is breaking a feature is not an issue but I don't insist.

**Reply by barbudor** [reply]
@kmgalanakis May be something got unconfigured while upgrading.
I have Sonoff Basic working with tasmota-lite 13.1.0.1 without any issue
Double check your configuration or provide de required details
Without any info, help is impossible

My own config is as per below:
```
11:56:24.703 CMD: Backlog Template; Module; GPIO 255
11:56:24.777 MQT: stat/htr-smallroom/TEMPLATE = {"NAME":"Sonoff Basic","GPIO":[32,1,1,1,1,0,0,0,224,576,288,0,0,0],"FLAG":0,"BASE":18}
11:56:25.003 MQT: stat/htr-smallroom/MODULE = {"Module":{"0":"Sonoff Basic"}}
11:56:25.306 MQT: stat/htr-smallroom/GPIO = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"576":"LedLink_i"},"GPIO14":{"288":"Led1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
```

And I insist, you should use tasmota-lite on that device

**Reply by sfromis** [reply]
You did not fill in the issue template to make it possible to see what happened. Thus, zero signs of something needing fixing in Tasmota as such.

**Reply by kmgalanakis** [reply]
how can I extract this information @barbudor and @sfromis?

**Reply by kmgalanakis** [reply]
I've updated the description with all the required information

**Reply by kmgalanakis** [reply]
@barbudor I've applied your template and have the exact same config as you, but still button not working 

```
13:29:53.805 CMD: Backlog Template; Module; GPIO 255
13:29:53.879 MQT: stat/sonoff3/RESULT = {"NAME":"Sonoff Basic","GPIO":[32,1,1,1,1,0,0,0,224,576,288,0,0,0],"FLAG":0,"BASE":18}
13:29:54.093 MQT: stat/sonoff3/RESULT = {"Module":{"0":"Sonoff Basic"}}
13:29:54.298 MQT: stat/sonoff3/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"576":"LedLink_i"},"GPIO14":{"288":"Led1"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}
```

**Comment by kmgalanakis** [comment, marked as answer]
I tried to reset the configuration but when the sonoff was back up I was unable to connect it to my wifi. So I downgraded to 12.3.1, reset the configuration and set up the sonoff from the beginning and finally, I updated to 13.1.0. Now everything works fine.

**Reply by barbudor** [reply]
Something may have been corrupted in the settings during the upgrade process
A good refresh is foften usefull
